app/routes/download_plot.py

```diff
@@ -0,0 +1,43 @@
+import io
+from matplotlib.backends.backend_agg import FigureCanvasAgg
+from flask import Response
+import redis
+from rq import Connection, Queue
+
+from app import app
+from app.utils.make_figure import make_figure
+
+from config import Configuration
+
+config = Configuration()
+
+@app.route('/download_plot/<string:job_id>', methods=['GET'])
+def download_plot(job_id=None):
+
+    """Returns the plot with the scores"""
+
+    with io.BytesIO() as output:
+        if job_id is not None:
+            redis_url = Configuration.REDIS_URL
+            redis_conn = redis.from_url(redis_url)
+            with Connection(redis_conn):
+                q = Queue(name=Configuration.QUEUE)
+                task = q.fetch_job(job_id)
+
+            # get the result as json
```

Comment:
What happens if the result is not found? For example, if the requested job ID does not exist, or if the result is not ready yet? I think the best thing is to return a different HTTP status code for these cases.

Reply:
This change is, of course, not required for completing the task (it was also not done in the original `classification` functionalities if I recall correctly). However, it would be a great chance for you to think about these problems in advance 👍